### PR TITLE
feat: Add API to provide icons for authentication providers

### PR DESCRIPTION
**Pull request by dgolovin**
### What does this PR do?

It adds API to configure authentication provider icons to show them in authentication settings UI

### Screenshot/screencast of this PR

When an Icon is configured for provider

image

When an icon is not configured the default icon is shown

image

### What issues does this PR fix or reference?

N/A

### How to test this PR?

N/A

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
It is missing a unit test for register authentication provider with images field ( check with different type of images with or without theme)
Else LGTM

**Comment by dgolovin** [comment]
@benoitf API is just passing it through to UI. I'll try to add tests on UI side.

**Comment by benoitf** [comment]
@dgolovin the stuff that you added in extension-loader needs to be tested. It is not only 'passing' data. Like checking if image is correctly updated to a base64 version, etc.

**Comment by dgolovin** [comment]
@benoitf that is correct observation, i missed that. Will check what I can do before merging it.

**Comment by jeffmaury** [comment]
The code LGTM but I tried to test it and can't make it work: when I go to the Authentication page, the icon is not displayed. I used this PR and the main branch from the red hat account extension

**Review by jeffmaury — APPROVED** [review]
LGTM

**Review by cdrage — APPROVED** [review]
Works and I'm able to see the images appearing.

Missing unit test (as mentioned before), but LGTM.

**Comment by dgolovin** [comment]
Added tests to cover updateImage() and registerAuthenticationProvider() methods.

**Pull request merged**
